### DISCUSSION TITLE
Imports refactoring on MOVE/RENAME of the entire package

### DISCUSSION THREAD

**Discussion by funkindy** [microsoft/pylance-release#5813] [Enhancement]
I am doing structural refactoring of my project and would like to move a package called `base` to the package called `common`.

Before the refactoring they are on the same level:

```
src/
├─ base/
├─ common/
```

I tried two options:
* Drag and drop `base` into `common`
* Rename `base` to `common/base`
each options leads to the desired structure:

```
src/
├─ base/
│  ├─ common/
```

But there is no import refactoring throughout the project after that, so i manually have to change `from base.*` to `from common.base.*`

it would be a nice feature to auto refactor import on module/package move.

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by heejaechang** [comment]
I guess, currently we only support rename folder, not move folder

**Comment by stephenkraemer** [comment]
In my hands, folder rename refactoring is broken, as well as rename refactoring when moving multiple files, as already reported by someone else: #5310

Having full support for file and folder move refactoring would be a huge help for many developers, thank you for considering it!

**Reply by tengerye** [reply]
It worked months ago, but I think it is broken again now.
